pageserver/src/tenant/ephemeral_file.rs

```diff
@@ -166,7 +168,7 @@ impl BlobWriter for EphemeralFile {
                                         error!("ephemeral_file write_blob failed to get immutable buf to pre-warm page cache: {e:?}");
                                         // fail gracefully, it's not the end of the world if we can't pre-warm the cache here
                                     }
-                                }
+                                }*/
```

Comment:
Why is this commented out?

Reply:
It's accessing the page cache in the write path, to prewarm it. As the comment at the top says, this is a good idea to do for performance but it's not required for correctness.

With this PR, page cache access requires `async`, which would require us to at least make the `write_blob` function async. 

We can remove this prewarming entirely, although an initial survey of the call graph seems to be promising: it seems to not be used in that many instances, and it might be possible to just convert this single `write_blob` function by moving it out of the trait without touching the other implementors (unlike for `BlockReader` where a lot of places want to be generic over it). That could either be some preliminary work for this PR, or we could comment it out for now and do it later. I will take a look into how complicated it is.

Reply:
It turns out it was fairly simple to make async: https://github.com/neondatabase/neon/pull/5056